src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs

```diff
@@ -865,7 +863,7 @@ private sealed class SyncStream : Stream, IDisposable
         {
             private readonly Stream _stream;
 
-            internal SyncStream(Stream stream) => _stream = stream ?? throw new ArgumentNullException(nameof(stream));
+            internal SyncStream(Stream stream!!) => _stream = stream;
```

Comment:
source.dot.net claims that there's only one caller of this ctor (lines 722-723) and it is already guaranteeing non-null.

If we wanted, we could probably change this to Debug.Assert.

Reply:
Makes sense.  I'll change that up subsequently (unless I need to restart CI, in which case I'll do it here).